extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -66,27 +66,20 @@ public static function onAfterInitialize( Title $title, $article, OutputPage $ou
 	}
 
 	/**
-	 * Handle downgrading anonymous requests for our sitemaps.
+	 * Handle downgrading anonymous requests for our sitemaps and robots.txt.
 	 *
-	 * @param  string     $subpage Specific sitemap being requested.
 	 * @param  WebRequest $request
 	 * @param  User       $user
 	 * @return boolean
 	 */
-	public static function onSitemapPageBeforeOutput( string $subpage, WebRequest $request, User $user ): bool {
-		global $wgScriptPath;
-		return self::handleRedirectForPath( "$wgScriptPath/$subpage", $request, $user );
-	}
-
-	/**
-	 * Handle downgrading anonymous requests for robots.txt.
-	 *
-	 * @param  WebRequest $request
-	 * @param  User       $user
-	 * @return boolean
-	 */
-	public static function onWikiaRobotsBeforeOutput( WebRequest $request, User $user ): bool {
-		return self::handleRedirectForPath( '/robots.txt', $request, $user );
+	public static function onSitemapRobotsPageBeforeOutput( WebRequest $request, User $user ): bool {
+		if ( WebRequest::detectProtocol() === 'https' &&
```

Comment:
This is no longer just downgrading after https://wikia-inc.atlassian.net/browse/PLATFORM-3552 which is why the method below was renamed to just `handleRedirectForPath`, so having this check here undoes the work in PLATFORM-3552 for robots and sitemaps. I think this hook should now just be the content of what was the `handleRedirectForPath` method below.

Reply:
Oops. Fixed that in 017ce92037c